sw/airborne/modules/computer_vision/snake_gate_detection.c

```diff
@@ -275,22 +276,22 @@ int snake_gate_detection(struct image_t *img, int n_samples, int min_px_size, fl
       check_gate_outline(img, temp_check_gate, &temp_check_gate.quality, &temp_check_gate.n_sides);
 
       // If the gate is good enough:
-      if (temp_check_gate.n_sides > min_n_sides && temp_check_gate.quality > best_gate.quality) {
+      if (temp_check_gate.n_sides >= min_n_sides && temp_check_gate.quality > (*best_gate).quality) {
         // store the information in the gate:
-        best_gate.x = temp_check_gate.x;
-        best_gate.y = temp_check_gate.y;
-        best_gate.sz = temp_check_gate.sz;
-        best_gate.sz_left = temp_check_gate.sz_left;
-        best_gate.sz_right = temp_check_gate.sz_right;
-        best_gate.quality = temp_check_gate.quality;
-        best_gate.n_sides = temp_check_gate.n_sides;
-        memcpy(best_gate.x_corners, temp_check_gate.x_corners, sizeof(int) * 4);
-        memcpy(best_gate.y_corners, temp_check_gate.y_corners, sizeof(int) * 4);
+        (*best_gate).x = temp_check_gate.x;
+        (*best_gate).y = temp_check_gate.y;
+        (*best_gate).sz = temp_check_gate.sz;
+        (*best_gate).sz_left = temp_check_gate.sz_left;
+        (*best_gate).sz_right = temp_check_gate.sz_right;
+        (*best_gate).quality = temp_check_gate.quality;
+        (*best_gate).n_sides = temp_check_gate.n_sides;
+        memcpy(&((*best_gate).x_corners[0]), &(temp_check_gate.x_corners[0]), sizeof(int) * 4);
+        memcpy(&((*best_gate).y_corners[0]), &(temp_check_gate.y_corners[0]), sizeof(int) * 4);
```

Comment:
strange that it is back in the old form... I have a habit of putting things off till tomorrow when they do not have to be done today, since there are other pressing deadlines tomorrow that need to be solved today. But I will do it.

Reply:
Hahahaha